src/ray/object_manager/plasma/stats_collector.cc

```diff
@@ -176,16 +208,29 @@ int64_t ObjectStatsCollector::GetNumBytesCreatedCurrent() const {
 }
 
 void ObjectStatsCollector::RecordMetrics() const {
+  // Shared memory sealed
+  ray::stats::STATS_object_store_memory.Record(
+      bytes_by_loc_seal_.Get({/* fallback_allocated */ false, /* sealed */ true}),
+      {{ray::stats::LocationKey.name(), ray::stats::kObjectLocMmapShm},
+       {ray::stats::ObjectStateKey.name(), ray::stats::kObjectSealed}});
+
+  // Shared memory unsealed
   ray::stats::STATS_object_store_memory.Record(
-      GetNumBytesCreatedCurrent() - num_bytes_unsealed_,
-      {{ray::stats::LocationKey.name(), ray::stats::kObjectLocInMemory}});
+      bytes_by_loc_seal_.Get({/* fallback_allocated */ false, /* sealed */ false}),
+      {{ray::stats::LocationKey.name(), ray::stats::kObjectLocMmapShm},
+       {ray::stats::ObjectStateKey.name(), ray::stats::kObjectUnsealed}});
 
+  // Fallback memory sealed
   ray::stats::STATS_object_store_memory.Record(
-      num_bytes_unsealed_,
-      {{ray::stats::LocationKey.name(), ray::stats::kObjectLocUnsealed}});
+      bytes_by_loc_seal_.Get({/* fallback_allocated */ true, /* sealed */ true}),
+      {{ray::stats::LocationKey.name(), ray::stats::kObjectLocMmapDisk},
+       {ray::stats::ObjectStateKey.name(), ray::stats::kObjectSealed}});
```

Comment:
I think you don't need the `.name()` here.

Reply:
Not with the current `Record` but add another one that skips a bunch of tag key register operations.